server/bin/pbench-tarball-stats.py

```diff
@@ -91,21 +91,49 @@ def get_sat_name(ctrl: str):
     return "main" if len(parts) == 1 else parts[0]
 
 
-def transform_buckets(buckets: dict):
+_sevendays = timedelta(days=7)
+
+
+def week_end(dt: datetime) -> str:
+    """Return a string representing the week ending date of the given datetime
+    object.
+    """
+    week_end = dt - timedelta(days=dt.isoweekday()) + _sevendays
+    return week_end.strftime("%Y-%m-%d")
```

Comment:
I've started just using f-strings for this, because it's a bit easier to read `f"{week_end:%Y-%m-%d}"` and it's rather cool to be able to do that. 😁